src/vs/workbench/contrib/terminal/browser/terminalEditor.ts

```diff
@@ -238,13 +238,21 @@ export class TerminalEditor extends EditorPane {
 			}
 		}
 
-		// TODO: Support contributed profiles with editor target
 		for (const contributed of this._terminalContributionService.terminalProfiles) {
-			dropdownActions.push(new Action(TerminalCommandId.NewWithProfile, contributed.title.replace(/[\n\r\t]/g, ''), undefined, true, () => this._terminalService.createContributedTerminalProfile(contributed.extensionIdentifier, contributed.id, {
-				isSplitTerminal: false,
+			dropdownActions.push(new Action(TerminalCommandId.NewWithProfile, contributed.title.replace(/[\n\r\t]/g, ''), undefined, true, () => this._terminalService.createTerminal({
+				config: {
+					extensionIdentifier: contributed.extensionIdentifier,
+					id: contributed.id,
+					title: contributed.title.replace(/[\n\r\t]/g, '')
+				},
 				target: TerminalLocation.Editor
 			})));
-			submenuActions.push(new Action(TerminalCommandId.NewWithProfile, contributed.title.replace(/[\n\r\t]/g, ''), undefined, true, () => this._terminalService.createContributedTerminalProfile(contributed.extensionIdentifier, contributed.id, {
+			submenuActions.push(new Action(TerminalCommandId.NewWithProfile, contributed.title.replace(/[\n\r\t]/g, ''), undefined, true, () => this._terminalService.createTerminal({
+				config: {
+					extensionIdentifier: contributed.extensionIdentifier,
+					id: contributed.id,
+					title: contributed.title.replace(/[\n\r\t]/g, '')
+				},
```

Comment:
This should use splitInstance using the current model, see https://github.com/microsoft/vscode/issues/129267

Reply:
it is already - maybe the above is outdated? https://github.com/microsoft/vscode/blob/a1b4b8d30316ad42f12d5d94a153800106de1a76/src/vs/workbench/contrib/terminal/browser/terminalEditor.ts#L250-L258

Reply:
`this._terminalService.splitInstance(...)` I mean